File: apps/notification-service/app/models/snapshot.py
from pydantic import BaseModel, Field
from bson import ObjectId
from typing import Optional
from datetime import datetime
from app.models.mongo import PyObjectId


class Temperature(BaseModel):
    inside: float
    outside: float
    casing: float
    top: float
    bottom: float


class Voltage(BaseModel):
    in_: float = Field(..., alias="in")
    battery: float


class Snapshot(BaseModel):
    id: Optional[PyObjectId] = Field(default_factory=PyObjectId, alias="_id")
    slug: str
    temperature: Temperature
    voltage: Voltage
    version: int
    timestamp: datetime

    class Config:
        json_encoders = {ObjectId: str}
        allow_population_by_field_name = True

File: apps/babybox-service/internal/domain/snapshot.go
package domain

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Snapshot struct {
	ID          primitive.ObjectID `json:"-"`
	Slug        string             `json:"slug"`
	Temperature Temperature        `json:"temperature"`
	Voltage     Voltage            `json:"voltage"`
	Version     uint               `json:"version"`
	Timestamp   string             `json:"timestamp"`
}

type Temperature struct {
	Inside  float64 `json:"inside"`
	Outside float64 `json:"outside"`
	Casing  float64 `json:"casing"`
	Top     float64 `json:"top"`
	Bottom  float64 `json:"bottom"`
}

type Voltage struct {
	In      float64 `json:"in"`
	Battery float64 `json:"battery"`
}
